diff --git a/lisp/comp/comp.l b/lisp/comp/comp.l
index abf7b7b..60c0ab5 100644
--- a/lisp/comp/comp.l
+++ b/lisp/comp/comp.l
@@ -1226,6 +1226,7 @@
 		 (verbose *verbose*)
 		 (entry (lisp::gencname-tail (pathname-name file)))
 		 (o))
+#-(and :arm :word-size=32) ;; armhf does not work with -O2, see https://github.com/euslisp/EusLisp/pull/427
    (if (and (integerp c-optimize)
 	    (> c-optimize 0)
 	    ;; (not (member :gcc *features*))
diff --git a/lisp/comp/trans.l b/lisp/comp/trans.l
index 3088d63..efe2e66 100644
--- a/lisp/comp/trans.l
+++ b/lisp/comp/trans.l
@@ -125,7 +125,7 @@
     (format cfile "register context *ctx;~%")
     (format cfile "register int n; register pointer argv[]; pointer env;~%")
     (format cfile "{ register pointer *local=ctx->vsp, w, *fqv=qv;~%")
-    (format cfile "  numunion nu;~%")
+    (format cfile "  volatile numunion nu;~%") ;; arm have optimization problem with numunion see https://github.com/euslisp/EusLisp/pull/427
     )
  (:check-req-arg (req opt)	;check number of required arguments
    (if (> *safety* 1) (format cfile "	breakck;~%"))
